Modules/IO/MINC/src/itkMINCImageIO.cxx

```diff
@@ -1136,9 +1136,8 @@ MINCImageIO::WriteImageInformation()
             miget_dimension_size(m_MINCPImpl->m_MincApparentDims[j], &_sz);
 
             _start = _start + (_sz - 1) * _sep;
-            _sep = -_sep;
 
-            miset_dimension_separation(m_MINCPImpl->m_MincApparentDims[j], _sep);
+            miset_dimension_separation(m_MINCPImpl->m_MincApparentDims[j], -_sep);
```

Comment:
Just to clarify: Of course, this is just a very minor improvement to me. I prefer not to unnecessarily modify variables (if possible), just like I prefer const over non-const variables. So I do prefer not to modify `_sep` at this point, and to pass `-_sep` as argument. But the _actual_ reason why this particular case bothered me was because it fooled me when I was searching for lines of code of the form `T * p; p = x` (pull request #4937). The regular expression I used (`^( [ ]+)([^ ].*\*)([ ]+)(\w+);[\r\n]+\1\4\ = `) accidentally matched with those two lines of code:

     _start = _start + (_sz - 1) * _sep;
     _sep = -_sep;

So it treated the first line of code as a declaration of a pointer, named `_sep`! Which was silly, of course! But it bothered me, nevertheless!

Reply:
Perhaps adding a comment "to make sign flip clearly visible" would be appropriate?

Reply:
@blowekamp Do you mean the following?
```cpp
_start = _start + (_sz - 1) * _sep;
// to make sign flip clearly visible
_sep = -_sep;
```

That line of comment does keep my regular expression for `T * p; p = x` away for a little while. But then later, someone may think of a clever regular expression to bypass the comment. Which is not unreasonable to me:

```cpp
int * ptr;
// Let's make the pointer null!
ptr = nullptr;
``` 
Would ideally be rewritten to:
```cpp
// Let's make the pointer null!
int * ptr = nullptr;
```

Anyway, the main question is, when `_sep` is a variable, is the meaning of passing `-_sep` as argument clear enough to anyone? I think it is. 🤷

Reply:
this code flips sign of the step , it needs to adjust the start coordinate and then do actual sign and then store it in the internal variables.

Reply:
Thanks @vfonov With the change I'm proposing here, the local `_sep` variable simply keeps original value, as retrieved by calling the _get_ function `miget_dimension_separation`. We could rename the variable to `originalSeparator` to make that clearer. And then pass `-originalSeparator` to the _set_ function `miset_dimension_separation`:

    miset_dimension_separation(m_MINCPImpl->m_MincApparentDims[j], -originalSeparator);

 Would that be clear enough to you?

Reply:
Of course, if you are opposed to this style change, feel free to press "Request changes". No problem.

Reply:
@hjmjohnson In this particular case I think I just give up. The removal of `_sep = -_sep` would only be a small code improvement (in my opinion) anyway.

More generally speaking, I would prefer to only add an extra variable when the variable is used _multiple times_ afterwards. A variable that is only used once _usually_ doesn't appeal very much to me. Sometimes a well-named variable can slightly improve readability, but it also adds up to the number of lines of code, which might then become more complicated (depending on the scope of all those variables). But it's partially just a matter of taste 🤷 

Thanks for asking, Hans! 😃